### PR TITLE
ENH: add send decom image action

### PR DESCRIPTION
**Pull request by anish-mudaraddi**
Adds the send decom image script and action

### CODE REVIEW

**Review by khalford — DISMISSED** [review]
LGTM

File: lib/workflows/send_decom_image_email.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    ), "please provide a list of image names to decommission"
+
+    assert len(image_name_list) == len(
+        image_eol_list
+    ), "Lists un-equal: Each image to decommission requires an EOL date"
+
+    assert len(image_name_list) == len(image_upgrade_list), (
+        "Lists un-equal: Each image to decommission requires a recommended upgraded image to replace it, "
+        "leave blank '' to specify if no upgrade image exists (and deletion is recommended)"
+    )
```
Comment (DavidFair) [inline lib/workflows/send_decom_image_email.py:65]: These should be `if` statements, asserts are for programming errors which shouldn't happen

Instead this should be either a `ValueError` or something similar

File: lib/workflows/send_decom_image_email.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+    img_list = []
+    for image_name, eol, upgrade_image in zip(
+        image_name_list, image_eol_list, image_upgrade_list
+    ):
+        assert datetime.strptime(
+            eol, "%Y/%m/%d"
+        ), "End Of Life string must be in format YYYY/MM/DD"
```
Comment (DavidFair) [inline lib/workflows/send_decom_image_email.py:73]: ^

File: lib/workflows/send_decom_image_email.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+        else:
+            email_params = build_email_params(
+                user_name,
+                get_affected_images_plaintext(decom_image_info)
+                if not as_html
+                else get_affected_images_html(decom_image_info),
+                server_query.to_string(groups=[user_id])
+                if not as_html
+                else server_query.to_html(groups=[user_id], include_group_titles=False),
```
Comment (DavidFair) [inline lib/workflows/send_decom_image_email.py:276]: Nit: can we do these if-else outside of this method call as it's a bit difficult to read

**Review by DavidFair — CHANGES_REQUESTED** [review]

**Review by DavidFair — APPROVED** [review]

**Review by khalford — APPROVED** [review]

**Pull request merged**